GVRf/Framework/src/org/gearvrf/GVRMesh.java

```diff
@@ -272,13 +291,31 @@ public GVRMesh getBoundingBox() {
         return new GVRMesh(getGVRContext(), NativeMesh.getBoundingBox(getPtr()));
     }
 
-    private static void isValidVector(float[] vector, int expectedLength) {
-        if (vector == null) {
+    private static void isNonNull(Object data) {
+        if (data == null) {
             throw new IllegalArgumentException(
-                    "Input vector should not be null.");
-        } else if (expectedLength != vector.length) {
-            throw new IllegalArgumentException("Input vector should have "
-                    + expectedLength + " elements, not " + vector.length);
+                    "The input array should not be null.");
+        }
+    }
+
+    private static void isValidData(int dataLength, int expectedComponents) {
+        if (dataLength == 0 || dataLength % expectedComponents != 0) {
+            throw Exceptions
+                    .IllegalArgument(
+                            "The input array should be an array of %d-component elements whose length is non-zero and divisible by %d. But current data length is %d.",
+                            expectedComponents, expectedComponents, dataLength);
+        }
+    }
+
+    private void verticesCheck(int dataLength, int expectedComponents) {
+        int verticesNumber = getVertices().length / 3;
+        int numberOfElements = dataLength / expectedComponents;
```

Comment:
Hi Jon.

I think for this issue, the isValidData has already taken care of that so it should be fine.